eclipse/src/saros/editor/EditorManager.java

```diff
@@ -1326,7 +1343,9 @@ private void updateEditorPool() {
     IEditorPart activeEditor = EditorAPI.getActiveEditor();
 
     if (activeEditor != null) {
-      locallyActiveEditor = EditorAPI.getEditorFile(activeEditor);
+      IFile file = EditorAPI.getEditorResource(activeEditor).getAdapter(IFile.class);
+      locallyActiveEditor = convertToFile(file);
```

Comment:
As convertToFile can return null I guess the logic may act slightly different as before (have not looked at the null checks, to be more specific isShared(locallyActiveEditor) lines).

Reply:
`SarosSession.isShared(IResource)` delegates the call to [`SharedReferencePointMapper.isShared(IResource)`](https://github.com/saros-project/saros/blob/pr/eclipse/ref-point-migration/1/inner-classes/core/src/saros/session/internal/SharedReferencePointMapper.java#L140), which checks whether the input is `null` and just returns `false` in such cases.

I am not really sure what you mean by "slightly different behavior". This still sets the `locallyActiveEditor` to the currently active shared editor (if present). The only difference is that it no longer also holds the reference to non-shared active editors. But such editors should not be of interest to Saros. Furthermore, in the only corner case that I can think of where this could matter – the currently active, non-shared editor belonging to a resource that is added to the session – the variable is updated as part of the [`resourcesAdded(...)` listener in `EditorManager`](https://github.com/saros-project/saros/blob/pr/eclipse/ref-point-migration/1/inner-classes/eclipse/src/saros/editor/EditorManager.java#L326), calling [`updateEditorPool()`](https://github.com/saros-project/saros/blob/pr/eclipse/ref-point-migration/1/inner-classes/eclipse/src/saros/editor/EditorManager.java#L1327)

Reply:
The state of a non shared editor is also broadcasted. But I guess as we do not distinguish between no opened editor and a non shared editor I guess it will not matter.

Looking at the code and if I am not mistaken it was possible to send an SPath pointing to a non shared editor which should be normally null.

Reply:
> Looking at the code and if I am not mistaken it was possible to send an SPath pointing to a non shared editor which should be normally null.

Even if it was possible, it is no longer possible with the new `ResourceTransportWrapper` logic that replaced the `SPath`. The new logic now always requires Saros `IResource` objects which can only be created for shared resources.

It is, however, still possible to send `null` as the resource for the selected editor to tell the other clients that no shared editor is currently active.